### PR TITLE
initial implementation for GP Data

### PR DESCRIPTION
**Pull request by markysoft**

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

File: app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+requireEnv(['JSON_FILE_URL']);
+requireEnv(['ES_HOST']);
+
+async function runUpdater() {
+  // run on initial start, then on the schedule
+  await updateElasticSearch();
+
+  log.info(`Scheduling Elasticsearch update with rule '${config.updateSchedule}'`);
+  schedule.scheduleJob(config.updateSchedule, () => {
+    // this is an async function, but await is not allowed within a lambda expression
```
Comment (neilbmclaughlin) [inline app.js]: Lambda's are good for terseness but it doesn't preclude using a function if there are other factors such as conveying intent without comments.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  base-app:
+    build:
+      context: .
+    volumes:
+      - .:/code
+    environment:
+      NODE_ENV: ${NODE_ENV}
+      UPDATE_SCHEDULE: ${UPDATE_SCHEDULE}
+      JSON_FILE_URL: https://nhsukgpdataetl.blob.core.windows.net/etl-output/gp-data-merged.json
```
Comment (neilbmclaughlin) [inline docker-compose.yml:13]: Since this URL is the default setting in config is this needed?

File: config/profiles/transform.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+function toNameObject(name) {
+  return { name };
+}
+
+function sanitiseDataRecord(record) {
+  /* eslint-disable no-underscore-dangle*/
+  /* eslint-disable no-param-reassign*/
+  if (record._id) {
+    // not sure this is needed
+    // record.id = record._id;
```
Comment (neilbmclaughlin) [inline config/profiles/transform.js]: Agree, `record.id` isn't needed. Delete.

File: app.js (hunk 1 above)
Comment (markysoft) [reply app.js]: Just realised lambda isn't needed at all

File: docker-compose.yml (hunk 2 above)
Comment (markysoft) [reply docker-compose.yml:13]: The default was removed in a previous commit, so there is no longer a default.

**Review by neilbmclaughlin — APPROVED** [review]

**Comment by c2s-dev** [comment]
:rocket: deployment of elasticsearch-updater succeeded (http://elasticsearch-updater-pr-1.dev.beta.nhschoices.net)

**Pull request merged**